### PR TITLE
[ANCHOR-360] SEP-6: Implement GET transaction(s)

### PR DESCRIPTION
**Pull request by philipliu**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Implements the GET `/transaction` and `/transactions` endpoints for SEP-6.

### Why

This is required for SEP-6 integration and e2e testing.

### Known limitations

This is only covered by unit tests for now. The deposit PR will exercise these endpoints in integration tests.

### CODE REVIEW

File: api-schema/src/main/java/org/stellar/anchor/api/sep/sep6/RefundPayment.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.stellar.anchor.api.sep.sep6;
+
+import com.google.gson.annotations.SerializedName;
+import lombok.Builder;
+import lombok.Data;
+
+@Builder
+@Data
+public class RefundPayment {
```
Comment (lijamie98) [inline api-schema/src/main/java/org/stellar/anchor/api/sep/sep6/RefundPayment.java:9]: We have the class: `org.stellar.anchor.api.shared.RefundPayment`. Can we re-use?

File: api-schema/src/main/java/org/stellar/anchor/api/sep/sep6/Refunds.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package org.stellar.anchor.api.sep.sep6;
+
+import com.google.gson.annotations.SerializedName;
+import java.util.List;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class Refunds {
```
Comment (lijamie98) [inline api-schema/src/main/java/org/stellar/anchor/api/sep/sep6/Refunds.java:10]: We have the class: `package org.stellar.anchor.api.shared.Refunds`. Can we share?

File: api-schema/src/main/java/org/stellar/anchor/api/sep/sep6/Transaction.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
+package org.stellar.anchor.api.sep.sep6;
+
+import com.google.gson.annotations.SerializedName;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class Transaction {
```
Comment (lijamie98) [inline api-schema/src/main/java/org/stellar/anchor/api/sep/sep6/Transaction.java:9]: 💡 rename to `Sep6Transaction`? In a few cases, we need to work with the `Sep24Transaction`, `Sep31Transaction` and `Sep6Transaction` classes.

File: core/src/test/java/org/stellar/anchor/Sep6ServiceTest.kt [hunk 4]
```diff
@@ -122,4 +135,242 @@ class Sep6ServiceTest {
...
+        .kind("deposit")
+        .pagingId("1")
+        .lang("en-US")
+        .build()
+    val response = sep6Service.findTransactions(TestHelper.createSep10Jwt(TEST_ACCOUNT), request)
+
+    verify(exactly = 1) { txnStore.findTransactions(TEST_ACCOUNT, null, request) }
+
+    assert(response.transactions.size == 1)
+    assertEquals(depositTxn.id, response.transactions[0].id)
```
Comment (lijamie98) [inline core/src/test/java/org/stellar/anchor/Sep6ServiceTest.kt:288]: Can we use JSONAssert?

File: platform/src/main/java/org/stellar/anchor/platform/component/sep/SepBeans.java [hunk 5]
```diff
@@ -96,6 +97,8 @@ Sep38Config sep38Config() {
   public FilterRegistrationBean<Filter> sep10TokenFilter(JwtService jwtService) {
     FilterRegistrationBean<Filter> registrationBean = new FilterRegistrationBean<>();
     registrationBean.setFilter(new Sep10JwtFilter(jwtService));
+    registrationBean.addUrlPatterns("/sep6/transaction");
+    registrationBean.addUrlPatterns("/sep6/transactions");
```
Comment (lijamie98) [inline platform/src/main/java/org/stellar/anchor/platform/component/sep/SepBeans.java:101]: 💡 `/sep6/transactions*`?

File: core/src/test/java/org/stellar/anchor/Sep6ServiceTest.kt (hunk 4 above)
Comment (philipliu) [reply core/src/test/java/org/stellar/anchor/Sep6ServiceTest.kt:288]: I think this test is more robust as we are asserting the values in the transaction response came from the deposit transaction. With JSONAssert, we only assert that the transaction response fields have certain values.

File: api-schema/src/main/java/org/stellar/anchor/api/sep/sep6/RefundPayment.java (hunk 1 above)
Comment (philipliu) [reply api-schema/src/main/java/org/stellar/anchor/api/sep/sep6/RefundPayment.java:9]: Created ANCHOR-392 to use the shared type in SEP-24 and SEP-31 implementations.

**Review by lijamie98 — APPROVED** [review]
LGTM. Please feel free to merge when you feel comfortable.

**Pull request merged**